Source/JavaScriptCore/wasm/WasmFaultSignalHandler.cpp

```diff
@@ -38,12 +38,30 @@
 #include "WasmExceptionType.h"
 #include "WasmMemory.h"
 #include "WasmThunks.h"
+#include <wtf/CodePtr.h>
+#include <wtf/CryptographicallyRandomNumber.h>
 #include <wtf/HashSet.h>
 #include <wtf/Lock.h>
 #include <wtf/threads/Signals.h>
 
 namespace JSC { namespace Wasm {
 
+using WTF::OpaquePtrTag;
+using WTF::CodePtr;
+
+CodePtr<OpaquePtrTag> presignedTrampoline = { };
+
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
+MachExceptionSigningKey::MachExceptionSigningKey()
+{
+    randomSigningKey = WTF::cryptographicallyRandomNumber<uint32_t>();
+    // Sign the trampoline pointer using a random diversifier and stash it away before webcontent has started
+    presignedTrampoline = JSC::LLInt::getCodePtr<OpaquePtrTag>(wasm_throw_from_fault_handler_trampoline_reg_instance);
+    uint64_t diversifier = ptrauth_blend_discriminator((void *)(unsigned long)randomSigningKey, ptrauth_string_discriminator("pc"));
+    presignedTrampoline = CodePtr<OpaquePtrTag>::fromTaggedPtr(ptrauth_auth_and_resign(presignedTrampoline.taggedPtr(), ptrauth_key_process_independent_code, 0, ptrauth_key_process_independent_code, diversifier));
```

Comment:
I think this is going to crash on debug builds because `CodePtr<OpaquePtrTag>::fromTaggedPtr` will do `assertIsTaggedWith<tag>(ptr);`. As `OpaquePtrTag == NoPtrTag`, `assertIsTaggedWith` will check there's no tag bits.